### PR TITLE
[APG-454] Check if staff details exisit in db before fetching

### PR DESCRIPTION
**Pull request by anandelangovan1**
## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes UI for this change to work...
  - [ ] ... and they been released to production already

### Post-merge

- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/StaffService.kt [hunk 1]
```diff
@@ -15,24 +15,37 @@ class StaffService(
...
-    staffDetailResponse?.staffId?.let {
-      staffRepository.findByStaffId(it)
-        ?: return staffRepository.save(buildStaffEntity(staffDetailResponse))
+  fun fetchPomDetailsIfNotAlreadyExists(staffId: Int?, prisonNumber: String, pomType: PomType): StaffEntity? {
+    if (staffId == null) {
+      log.warn("No $pomType pom staffId found for $prisonNumber")
+      return null
+    }
+
+    val findByStaffId = staffRepository.findByStaffId(staffId)
```
Comment (pfcairns) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/StaffService.kt:39]: Naming : it's an odd name for a variable. Could you please rename.

**Review by pfcairns — COMMENTED** [review]
Curious as to why the id has changed from BigInteger to Int? I would have thought for a key it'd be safer to use BigInteger.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/StaffService.kt (hunk 1 above)
Comment (anandelangovan1) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/StaffService.kt:39]: That's like the default suggestion for a variable in intellij. Updated and pushed

**Comment by anandelangovan1** [comment]
> Curious as to why the id has changed from BigInteger to Int? I would have thought for a key it'd be safer to use BigInteger.

These are staffId which don't seem to be big enough for using BigInt and we are not doing any computations with it. I had BigInt initially when i dint know what to expect from moic api

**Pull request merged**